prac_08/convert_miles_km.py

```diff
@@ -0,0 +1,44 @@
+"""
+Estimated Time: 20
+Actual Time: 30
+"""
+
+from kivy.app import App
+from kivy.lang import Builder
+from kivy.core.window import Window
+from kivy.properties import StringProperty
+
+
+class ConvertMilesKm(App):
+    """Convert miles to km app."""
+    MILES_TO_KM_FACTOR = 1.60934
+    conversion_result = StringProperty()
+
+    def build(self):
+        """build the Kivy app from the kv file."""
+        Window.size = (800, 400)
+        self.title = "Convert Miles to Kilometres"
+        self.root = Builder.load_file('convert_miles_km.kv')
+        return self.root
+
+    def handle_conversion(self):
+        """Convert miles to km and store in conversion_results string."""
+        try:
+            miles = float(self.root.ids.text_input.text)
+            self.conversion_result = str(miles * self.MILES_TO_KM_FACTOR)
+        except ValueError:
+            self.conversion_result = '0.0'
+
+    def handle_increment(self, amount):
+        """Increment the input_field by a passed in amount."""
+        input_field_text = self.root.ids.text_input.text
+        try:
```

Comment:
Everything looks good here, my only suggestion would be considering making the error checking itself into a function, Lindsay's solution uses one for all handling of conversions :)

Reply:
I'll look into it as it might simplify the code and improve readability, thanks. 👍